Express nd

Edgaras318 opened the pull request:
https://github.com/Edgaras318/home-service-api/settings

Commit:
feat: Implement CRUD operations for Booking and Business

- Created API endpoints for managing bookings:
  - POST /api/bookings: Create a new booking
  - GET /api/bookings: Retrieve all bookings
  - GET /api/bookings/:id: Retrieve a specific booking by ID
  - PUT /api/bookings/:id: Update an existing booking
  - DELETE /api/bookings/:id: Delete a booking by ID

- Created API endpoints for managing businesses:
  - POST /api/businesses: Create a new business
  - GET /api/businesses: Retrieve all businesses
  - GET /api/businesses/:id: Retrieve a specific business by ID
  - PUT /api/businesses/:id: Update an existing business
  - DELETE /api/businesses/:id: Delete a business by ID

- Added error handling and input validation.
- Improved response messages for better clarity.
- Ensured proper status codes are returned for different scenarios.
- Connect to mongoDB

Commit:
feat: Update README with detailed API documentation

Expanded the README to include comprehensive API documentation. Added sections for features, technologies used, detailed setup, and usage instructions. Included specific endpoints and request/response examples for bookings, businesses, and categories.

Commit:
refactor: Simplify booking and business creation logic

Refactored booking and business creation endpoints to directly accept req.body. Added Joi validation schema for bookings and implemented a Mongoose pre-save hook for data validation. Included required dependencies and removed unnecessary comments.

Commit:
Add Swagger documentation for API routes

Added Swagger annotations to document the endpoints for businesses, categories, and bookings. Integrated Swagger UI into the server to serve the API documentation at `/api-docs`. Updated README to include instructions on accessing the Swagger docs.

Commit:
Add authentication and user management modules

Implemented routes for user login and registration, added middleware for token validation, and included JWT utility functions. Updated models to include user schema and password hashing, and modified env files with JWT secret configuration.

Commit:
Remove inline Swagger documentation and use external JSON

Removed inline Swagger documentation from routes and replaced it with an external `swagger.json` file. Updated `server.js` to read and parse the `swagger.json` file for API documentation setup. This change streamlines API documentation management and reduces clutter in route files.

Commit:
Add ESLint configuration and integrate with project

Added `eslint.config.mjs` to set up ESLint with common rules and plugins for the project. Adjusted various files to disable inline console logs per ESLint configuration. Included necessary ESLint packages in `package-lock.json` for proper linting capabilities.

Commit:
Add Prettier configuration and integrate across codebase
Add more eslint rules

Introduced `.prettierrc` for consistent code formatting with Prettier, enforcing single quotes, trailing commas, and a maximum print width of 120 characters. Additionally, updated existing files to align with the new Prettier rules, improving code readability and maintainability.

Commit:
Fix error handling and unify ESLint rule formatting

Added a missing return statement in the bookings controller to properly handle errors. Adjusted the ESLint config to use consistent formatting without quotes for object keys, ensuring better readability and compliance with linting standards.

Commit:
Enhance ESLint configuration with detailed rule comments

Added comments for existing ESLint rules to improve clarity and documentation. Introduced new rules to limit function complexity, enforce callback error handling, and disallow potentially hazardous JavaScript constructs.

Commit:
Update Prettier config and add .prettierignore file

Enabled bracket spacing in the Prettier configuration to improve code readability. Added a new .prettierignore file to exclude specific directories and files like node_modules and the ESLint config from Prettier formatting.

Commit:
Add authentication middleware to routes

Integrated an authentication middleware to secure the create, update, and delete operations for businesses, categories, and bookings routes. This ensures that only authenticated users can perform these actions.

ZilvinasVidmantas commented on .env:
```diff
@@ -0,0 +1,4 @@
+MONGO_URI=mongodb+srv://edgaras318:Edgaras1@cluster0.i1xgu.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0
+PORT=5001
+JWT_SECRET=d3aed2d849951ecf10896406658d23c06444d14a1be8cab84db325736ba3dde4
```
.env fails turi būti paslėptas su .gitignore.
ir sukurtas failas pavadinimu ".env.example" kuriame būtų kintamųjų pavadinimai be reikšmių

ZilvinasVidmantas reviewed (approved):
Net neturiu ką pasakyt, labai gerai

Commit:
Add .env to .gitignore

Includes .env file in the .gitignore to prevent sensitive information from being tracked in version control. This helps in maintaining security by keeping environment variables local and outside the repository.
